tools/cwrap/plugins/WrapDim.py

```diff
@@ -0,0 +1,45 @@
+from . import CWrapPlugin
+from string import Template
+
+
+class WrapDim(CWrapPlugin):
+
+    DIM_WRAP_NDIM_TEMPLATE = Template(
+        """${arg_tensor}->nDimension""")
+
+    DIM_WRAP_CHECK_TEMPLATE = Template(
+        """THPUtils_assert(${arg_to_wrap} >= -(${ndim}) && ${arg_to_wrap} < (${ndim}),
```

Comment:
it seems that `arg_to_wrap` is in fact `arg_dim` (i.e. it's already an unpacked number of dimensions).